### PR TITLE
M2P-553 Add support for MageWorx Advanced Product Options Suite

### PR DESCRIPTION
**Pull request by dumega**
# Description
With MageWorx Advanced Product Options Suite, the merchant can associate existing product as custom option, then the format of sku could combine with MAGENTO SKU DELIMITER.
eg. the sku of parent product is 24-MG01 and the skus of associated product options are 25-MG02 and 26-MG03, then the sku of cart item become 24-MG01-25-MG02-26-MG03 
as a result, the current logic to retrieve sku of parent product does not work anymore.

This PR mainly fix two issues:
1. Get the skus of associated product options properly.
2. Using str_replace(self::MAGENTO_SKU_DELIMITER.$customizableOption['sku'], '', $sku); to remove the custom option skus, cause that's the rule to build the sku of cart item which has custom options in Magento 2.

Fixes: https://boltpay.atlassian.net/browse/M2P-553

#changelog M2P-553 Add support for MageWorx Advanced Product Options Suite

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Review by njfaries — APPROVED** [review]
@danac-gs can you take a look? It looks fine to me but I'm not super sure for the systems involved here.

**Comment by dumega** [comment]
@danac-gs Please review when you get a chance.

File: Helper/Cart.php [hunk 1]
```diff
@@ -1540,29 +1540,30 @@ function ($item) use ($imageHelper, &$totalAmount, &$diff, $storeId, $currencyCo
     /**
      * Return the selected customizable options of quote item.
      *
-     * @param $product
+     * @param $item
```
Comment (sinisa-colic) [inline Helper/Cart.php:1543]:
```suggestion
     * @param Quote\Item $item
```

Comment (danac-gs) [inline Helper/Cart.php:1543]:
```suggestion
     * @param Quote/Item $item
```

File: Helper/Cart.php [hunk 2]
```diff
@@ -1540,29 +1540,30 @@ function ($item) use ($imageHelper, &$totalAmount, &$diff, $storeId, $currencyCo
...
         }
 
         $customizableOptions = [];
+        $product = $item->getProduct();
         foreach (explode(',', $optionIds->getValue()) as $optionId) {
             $option = $product->getOptionById($optionId);
             if ($option) {
                 $confItemOption = $product->getCustomOption(\Magento\Catalog\Model\Product\Type\AbstractType::OPTION_PREFIX . $optionId);
-
+                $itemOption = $item->getOptionByCode('option_' . $option->getId());
```
Comment (danac-gs) [inline Helper/Cart.php:1560]:
```suggestion
                $itemOption = $item->getOptionByCode(\Magento\Catalog\Model\Product\Type\AbstractType::OPTION_PREFIX . $option->getId());
```

**Review by danac-gs — APPROVED** [review]

File: Helper/Cart.php (hunk 1 above)
Comment (dumega) [reply Helper/Cart.php]: Done

File: Helper/Cart.php (hunk 2 above)
Comment (dumega) [reply Helper/Cart.php]: Done

**Pull request merged**